bsip-00XX Sharedrop Operation.md

```diff
@@ -0,0 +1,153 @@
+    BSIP: TBD
+    Title: Sharedrop operation
+    Authors: OpenLedgerApp <https://github.com/OpenLedgerApp>
+    Status: Draft
+    Type: Protocol
+    Created: 2018-12-17
+    Updated: 2018-12-17
+    Discussion: 
+    Worker: TBD
+ 
+A sharedrop is a giveaway of some tokens to a certain group of people, for example to all BTS holders in proportion to their holdings.
+
+# Abstract
+We should add the ability to distribute arbitrary amount of some token (SHAREDROP token) to holders of a specific other token (STAKE token)  according to the share(stake) of such tokens.
+The importance of a similar solution was raised in BSIPs-19, 20,  however never implemented. (link below).
+
+
+# Motivation
+By implementing this proposal, we let Asset Owners or other businesses an opportunity to distribute their profit to the holders of a specific asset in proportion of the owned amount of this asset.
+
+Also, usual token drops can be done via the sharedrop operation.
+
+# Rationale
+
+- Other cryptocurrency platforms offer profit-sharing/dividends functionality, such as Peerplays/NXT/CounterParty/DigixDAO/LBRY/Waves/Dash.
+- Most (if not all) Bitcoin forks (1000's of them) have 'sendmany' functionality which enables the mass distribution of their tokens against a large quantity of users (addresses) in a single transaction; this is currently not possible within BitShares.
```

Comment:
BitShares has a "sendmany" too. It is just a transaction with "many" transfer operations. Same thing that bitcoin forks do.

Reply:
We have looked at the way "sendmany" is implemented and it does not exactly cover our needs. It is like many different transfers.. Block can only have limited amount of transfers. So if there are too many, then we will have several transactions and it will not be atomic.

Reply:
The same will happen with an actual `send_many` operation. The thing that will make the transaction too big will be the `account_id` to `amount` list, and not the header of the serialization.

Reply:
A serialized `transfer_operation` in BitShares is much smaller than a bitcoin address. A BitShares transaction with many transfers will be significantly smaller than a bitcoin transaction with as many recipients.
For a large number of recipients you'll hit the bitcoin block size limit much faster than the BitShares block size limit. Not to mention the fees.